### ISSUE TITLE
Move `rstudioapi` and `rsconnect` from `Imports` to `Suggests`?

### ISSUE THREAD

**Issue by etiennebacher** [quarto-dev/quarto-r#134]
Hello, I have a package ([`altdoc`](https://github.com/etiennebacher/altdoc)) that uses the `quarto` package a lot. I don't need to publish documents via `rsconnect` and I also don't need `rstudioapi` given that this package mostly runs in CI. I'm wondering if it would be possible to move these two dependencies from `Imports` to `Suggests`.

From what I can see, you already check whether `rsconnect` is installed:

https://github.com/quarto-dev/quarto-r/blob/1c8dc29bf82767c4b46ea55ef5fc13494ead8328/R/publish.R#L386-L389

so moving it to `Suggests` shouldn't change the existing workflow of users.

Removing those two packages would reduce the number of dependencies from 44 to 36:

``` r
pak::pkg_deps("quarto", dependencies = "hard") |> 
  dplyr::pull(ref) |> 
  unique() |> 
  length()
#> ℹ Loading metadata database✔ Loading metadata database ... done
#> [1] 44
```

```r
# once rsconnect and rstudioapi are moved to Suggests
pak::local_deps(".", dependencies = "hard") |> 
    dplyr::pull(ref) |> 
    unique() |> 
    length()
#> [1] 36
```

Hard dependencies that would be removed:

```r
"askpass"    "curl"       "openssl"    "packrat"     "renv"       "rsconnect"  "rstudioapi" "sys"
```

**Comment by petergodbert** [comment]
Is there any way to see which next version this will be in?   Is it right to assume this means 1.4?

**Comment by cderv** [comment]
Yes next CRAN version will be 1.4 as currently 1.3 is the one released - https://cran.r-project.org/web/packages/quarto/index.html

I'll do a release soon after Quarto 1.4 release.
